tests/qmltests/tst_Shell.qml

```diff
@@ -3247,59 +3247,71 @@ Rectangle {
                 formFactor: "phone",
                 launcherLocked: false,
                 stateWithOpenIndicators: "",
-                stateWithOpenApps: ""
+                stateWithOpenApps: "",
+                inputMethodMargin: 0,
             },
             {
                 tag: "tablet",
                 shell: "tablet",
                 formFactor: "tablet",
                 launcherLocked: false,
                 stateWithOpenIndicators: "visible",
-                stateWithOpenApps: ""
+                stateWithOpenApps: "",
+                inputMethodMargin: 0,
             },
             {
                 tag: "desktop without auto-hide",
                 formFactor: "desktop",
                 shell: "desktop",
                 launcherLocked: true,
                 stateWithOpenIndicators: "visible",
-                stateWithOpenApps: "visible"
+                stateWithOpenApps: "visible",
+                inputMethodMargin: 64,
             },
             {
                 tag: "desktop with auto-hide",
                 formFactor: "desktop",
                 shell: "desktop",
                 launcherLocked: false,
                 stateWithOpenIndicators: "visible",
-                stateWithOpenApps: ""
+                stateWithOpenApps: "",
+                inputMethodMargin: 0,
             },
             {
                 tag: "desktop with auto-hide at small formFactor",
                 formFactor: "phone",
                 shell: "desktop",
                 launcherLocked: false,
                 stateWithOpenIndicators: "",
-                stateWithOpenApps: ""
+                stateWithOpenApps: "",
+                inputMethodMargin: 0,
             },
             {
                 tag: "tablet with auto-hide at small formFactor",
                 formFactor: "phone",
                 shell: "tablet",
                 launcherLocked: false,
                 stateWithOpenIndicators: "",
-                stateWithOpenApps: ""
+                stateWithOpenApps: "",
+                inputMethodMargin: 0,
             }]
         }
 
         /*
             Ensure that the Launcher will not dismiss while there are no apps
             running, but gets dismissed when it would be intersected by
             indicators.
+
+            Additionally, ensure that the keyboard takes up the full width of
+            the Shell when the launcher is locked by Shell functions and has
+            a margin set to avoid the Launcher when the launcher is locked by
+            the user.
         */
         function test_launcherLockedWhenNoAppsRunning(data) {
             loadShell(data.shell, 0);
             shellLoader.state = data.formFactor;
             var launcher = findChild(shell, "launcher");
+            var inputMethod = findChild(shell, "inputMethod")
```

Comment:
Should have a semicolon at the end here for consistency.

Reply:
Indeed, whoops. Fixed!